core/src/main/java/org/elasticsearch/index/query/InnerHitBuilder.java

```diff
@@ -94,7 +94,7 @@
                 ObjectParser.ValueType.OBJECT_ARRAY);
         PARSER.declareField((p, i, c) -> {
             try {
-                i.setFetchSourceContext(FetchSourceContext.parse(c));
+                i.setFetchSourceContext(FetchSourceContext.parse(c.parser(), c.getParseFieldMatcher()));
```

Comment:
`ParseFieldMatcher` is probably going to be removed. It doesn't really do its job any more so we're probably just going to pitch it. So don't feel bad about not passing it in and just using `ParseFieldMatcher.STRICT` at the call site.